test/src/main/java/org/apache/accumulo/test/functional/ManagerAssignmentIT.java

```diff
@@ -73,26 +77,43 @@ public static void beforeAll() throws Exception {
       cfg.setNumTservers(1);
       cfg.setProperty(Property.TSERV_ASSIGNMENT_MAXCONCURRENT, "10");
       cfg.setProperty(Property.GENERAL_THREADPOOL_SIZE, "10");
-      cfg.setProperty(Property.MANAGER_TABLET_GROUP_WATCHER_INTERVAL, "5");
+      cfg.setProperty(Property.MANAGER_TABLET_GROUP_WATCHER_INTERVAL, "5s");
+      cfg.setProperty(Property.TSERV_ONDEMAND_UNLOADER_INTERVAL, "10s");
+      cfg.setProperty("table.custom.ondemand.unloader.inactivity.threshold.seconds", "15");
     });
   }
 
   @Test
   public void test() throws Exception {
     try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      // Confirm that the root and metadata tables are hosted
+      Locations locs =
+          c.tableOperations().locate(RootTable.NAME, Collections.singletonList(new Range()));
+      locs.groupByTablet().keySet().forEach(tid -> assertNotNull(locs.getTabletLocation(tid)));
+
+      Locations locs2 =
+          c.tableOperations().locate(MetadataTable.NAME, Collections.singletonList(new Range()));
+      locs2.groupByTablet().keySet().forEach(tid -> assertNotNull(locs2.getTabletLocation(tid)));
+
       String tableName = super.getUniqueNames(1)[0];
       c.tableOperations().create(tableName);
+
       String tableId = c.tableOperations().tableIdMap().get(tableName);
-      // wait for the table to be online
+
+      // wait for the tablet to exist in the metadata table. The tablet
+      // will not be hosted so the current location will be empty.
       TabletLocationState newTablet;
       do {
         UtilWaitThread.sleep(250);
         newTablet = getTabletLocationState(c, tableId);
-      } while (newTablet.current == null);
+      } while (newTablet.extent == null);
```

Comment:
It would be a follow on issue.  These loops could be shortened using the new o.a.a.test.util.Wait class.

Reply:
Captured in comment of follow-on tasks